frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/repository/RepositoryGraph.java

```diff
@@ -83,18 +78,46 @@ public Integer getNodeIndex(int nodeId) {
     @SuppressWarnings("index:argument.type.incompatible")
     IGraphElement currentElement = elements.get(nodeIndex);
 
-    if (filter.downNormal && nodeIndex < elements.size() - 1) {
-      currentElement.getDownElementIndexes().stream()
-          .map(i -> GraphEdge.createNormalEdge(nodeIndex, i, GraphEdgeType.USUAL)).forEach(adjacentEdges::add);
-    }
-
     if (filter.upNormal && nodeIndex > 0) {
       int upIndex = currentElement.getUpElementIndex();
       if (upIndex >= 0) {
         adjacentEdges.add(GraphEdge.createNormalEdge(nodeIndex, upIndex, GraphEdgeType.USUAL));
       }
     }
 
+    if (filter.downNormal && nodeIndex < elements.size() - 1) {
```

Comment:
Hmm in general: is `node` and `element` the same or a different thing? :thinking:

Reply:
good point!

Reply:
renamed :party-parrot: